### PR TITLE
Allow custom data attributes

### PR DESCRIPTION
**Pull request by 10chars**
Our project required passing custom data-* values to the html node (instead of id and value).

This update attaches all values to the element dataset and allows you to select which of these to filter these via the `dataAttributes` options.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +73,7 @@ const quill = new Quill('#editor', {
 | `isolateCharacter`   | `false`        | Whether or not the denotation character(s) should be isolated. For example, to avoid mentioning in an email.
 | `fixMentionsToQuill` | `false`        | When set to true, the mentions menu will be rendered above or below the quill container. Otherwise, the mentions menu will track the denotation character(s);
 | `defaultMenuOrientation` | `'bottom'` | Options are `'bottom'` and `'top'`. Determines what the default orientation of the menu will be. Quill-mention will attempt to render the menu either above or below the editor. If `'top'` is provided as a value, and there is not enough space above the editor, the menu will be rendered below. Vice versa, if there is not enough space below the editor, and `'bottom'` is provided as a value (or no value is provided at all), the menu will be rendered above the editor.
+| `dataAttributes` | `null`        | A list of data values you wish to be passed from your list data to the html node. (`id, value, denotationChar` are included by default).
```
Comment (MadSpindel) [inline README.md]:
```suggestion
| `dataAttributes` | `['id', 'value', 'denotationChar', 'link']`        | A list of data values you wish to be passed from your list data to the html node. (`id, value, denotationChar, link` are included by default).
```

**Comment by 10chars** [comment]
Thanks, I have updated the readme with the suggested changes.  
Wasn't sure on the naming of the option `dataAttributes`, but if you're happy with it I will leave it as is.

**Comment by shyamal890** [comment]
+1

**Comment by 10chars** [comment]
> It would be great if we could have custom html instead.

Can you give me an example of what you would like to do? I can take a look

**Comment by 10chars** [comment]
> It would be great if we could have custom html instead.

Can you give me an example of what you would like to do? I can take a look

**Comment by 10chars** [comment]
> It would be great if we could have custom html instead.

Can you give me an example of what you would like to do? I can take a look

**Comment by 10chars** [comment]
> It would be great if we could have custom html instead.

Can you give me an example of what you would like to do? I can take a look

**Comment by shyamal890** [comment]
@10chars something where the pasted html can be a template or a passed string liked

`<a href="+item.link+" class="custom-mention" item_id="+item.id+">item.value</a>`

**Comment by 10chars** [comment]
Thanks for the example @shyamal890
Just a quick note `item_id` is invalid html. Data should be held in standard html data-attributes.

Do you have a specific use case in mind when wanting custom html?  e.g non inline etc

With the changes in this PR included, users already have the following options available for customization.

1. Override styles using the `mention` class for custom styling
2. Specify whether or not to wrap mentions in an `a tag` by providing the link field in the data.
3. Specify any number of custom attributes

One addition I can see that might be useful is to give users a custom `on-click` option. custom html maybe a little overkill unless you would like people to use this plugin for something other than standard mentions.

Any thoughts?

**Comment by shyamal890** [comment]
My point is custom html would just open up the possibilities for developers. Think in Angular/React having something like `<a href="something">Something<custom-element></custom-element></a>`

**Comment by shyamal890** [comment]
I think having custom html opens up possibilities for developers to do what they want. Apply their own classes, create custom elements, etc.

**Comment by 10chars** [comment]
@shyamal890 I'm happy to look at add templating option, but I think you have to consider the plugin objectives, i.e - mentioning users

For me, the expected functionality for a mentions/hashtags plugin is fairly straight forward, select one string value from a list of options and insert that content into the content as `inline` text.  

If people want to create "custom elements", they don't need a mention plugin, they need an `insert html element` plugin.

**Comment by MadSpindel** [comment]
I agree with you @10chars. I will try to merge this tomorrow. Thank you for your contribution!

**Pull request merged**